### PR TITLE
added support for string array

### PR DESCRIPTION
**Pull request by DvirDukhan**

### CODE REVIEW

File: utils.go [hunk 1]
```diff
@@ -3,18 +3,17 @@ package redisgraph
 import (
 	"crypto/rand"
 	"fmt"
-	"reflect"
 	"strings"
 )
 
 // go array to string is [1 2 3] for [1, 2, 3] array
 // cypher expects comma separated array
 func arrayToString(arr interface{}) interface{} {
```
Comment (swilly22) [inline utils.go]: Shouldn't this be
`func arrayToString(arr []interface{}) interface{} {`

File: utils.go [hunk 2]
```diff
@@ -23,10 +22,10 @@ func ToString(i interface{}) interface{} {
...
 	}
-	v := reflect.ValueOf(i)
-	switch reflect.TypeOf(i).Kind() {
-	case reflect.String:
-		s := v.String()
+
+	switch i.(type) {
+	case string:
+		s := i.(string)
 		if len(s) == 0 {
```
Comment (swilly22) [inline utils.go]: Shouldn't we wrap s in `" s "` regardless if s is: `"..."` e.g. s value contains `"`

File: utils.go [hunk 3]
```diff
@@ -3,18 +3,17 @@ package redisgraph
...
 // cypher expects comma separated array
 func arrayToString(arr interface{}) interface{} {
-	v := reflect.ValueOf(arr)
-	var arrayLength = v.Len()
+	v := arr.([]interface{})
+	var arrayLength = len(v)
 	strArray := []string{}
 	for i := 0; i < arrayLength; i++ {
-		strArray = append(strArray, fmt.Sprintf("%v", ToString(v.Index(i))))
+		strArray = append(strArray, fmt.Sprintf("%v", ToString(v[i])))
```
Comment (swilly22) [inline utils.go]: I think what we want here is to wrap `v[i]` in quotes if `v[i]` is of type string,
if this is indeed the case, use a different function, e.g. `QuoteString(v[i])`

Comment (DvirDukhan) [reply utils.go]: The `ToString` method does it

File: utils.go (hunk 1 above)
Comment (DvirDukhan) [reply utils.go]: changed

**Pull request merged**